src/Symfony/Component/HttpClient/Response/TraceableResponse.php

```diff
@@ -129,9 +172,23 @@ public static function stream(HttpClientInterface $client, iterable $responses,
 
             $traceableMap[$r->response] = $r;
             $wrappedResponses[] = $r->response;
+            if ($r->event && !$r->event->isStarted()) {
+                $r->event->start();
+            }
         }
 
         foreach ($client->stream($wrappedResponses, $timeout) as $r => $chunk) {
+            if ($traceableMap[$r]->event && $traceableMap[$r]->event->isStarted()) {
+                // Display a "lap" instead of "stop" in timeout only when we are able to revert the didThrow flag
+                if ($chunk instanceof ErrorChunk && $chunk->isTimeout()) {
```

Comment:
`instanceof ErrorChunk` is coupling to an implementation detail
try/catch to the rescue

Reply:
The instanceof is to check if we can revert the didThrow flag (see comment above).

I wanted to show a "lap" instead of a "stop" only if we can do it in a safe way.